### PR TITLE
Update picat to 2.6.2

### PR DESCRIPTION
**Pull request by boltomli**
The build version is 2.6#2

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
Re-releasing a new version of the software, under the same name, and erasing the original file is simply not going to work for us. If `picat` does not have stable released versions, we might have to remove it from Homebrew.

**Comment by zbeekman** [comment]
@BrewTestBot test this please

**Comment by zbeekman** [comment]
@BrewTestBot test this please

File: Formula/picat.rb [hunk 1]
```diff
@@ -25,6 +25,8 @@ def install
   end
 
   test do
+    ver = shell_output("#{bin}/picat --version").chomp
+    assert_equal "Picat version 2.6#2", ver
```
Comment (zbeekman) [inline Formula/picat.rb]: Please get rid of this new "test".

The goal of tests is to not have to update them with every new upstream release.

You're welcome to add additional testing that tests the software functions correctly, just like the one below.

**Review by zbeekman — CHANGES_REQUESTED** [review]

File: Formula/picat.rb (hunk 1 above)
Comment (zbeekman) [reply Formula/picat.rb]: ping @boltomli

Comment (zbeekman) [reply Formula/picat.rb]:
```suggestion
```

File: Formula/picat.rb [hunk 2]
```diff
@@ -25,6 +25,8 @@ def install
   end
 
   test do
+    ver = shell_output("#{bin}/picat --version").chomp
```
Comment (zbeekman) [inline Formula/picat.rb]:
```suggestion
```

**Comment by zbeekman** [comment]
@boltomli: I made the fixes on your branch. I'm a bit puzzled why picat chose a version name of `2.6#2`. Patch versions are pretty standard, so in the future I would recommend `2.6.2` as the version name for upstream...

**Review by zbeekman — APPROVED** [review]

**Comment by zbeekman** [comment]
Thanks for your contribution to Homebrew! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**